src/java/org/apache/cassandra/service/DefaultFSErrorHandler.java

```diff
@@ -50,7 +50,6 @@ public void handleCorruptSSTable(CorruptSSTableException e)
 
         switch (DatabaseDescriptor.getDiskFailurePolicy())
         {
-            case die:
```

Comment:
CASSANDRA-18294 added this?  Why are we removing it?

Reply:
Yeah, glad you noticed - that's because CC already added a `die` case in STAR-13, just below

```
            case die:
                JVMStabilityInspector.killCurrentJVM(e, false);
                break;
```

Should we keep the CC version or use the C* version?

Reply:
We should use the C* version if it’s the same end result. Less divergence the better.

Reply:
So I found that taking the OSS C* changes to `DefaultFSErrorHandler` caused test failures in four test classes that are expecting `JVMStabilityInspector.killCurrentJVM` to be called. Those tests are: `FailingRepairTest`, `CompactionOutOfSpaceTest`, `DiskFailurePolicyTest` and `OutOfSpaceTest`.

Checking those tests locally, I found using the CC version fixed the problem, and I wanted to see the result of a full build using that (and other fixes I’ve been making).

Using the CC version changes, there was just one test failure, the new DefaultFSErrorHandlerTest added in CASSANDRA-18294.

Here’s the latest butler:
http://butler-stargazer.aws.dsinternal.org/#/ci/upstream/compare/ds-cassandra-pr-gate/oss-4.0.11-rebase-cc-rebase-squash

I think the root of this situation is not really with DefaultFSErrorHandler itself, but the differences in JVMStabilityInspector between OSS and CC … I remember that being something during the porting efforts on CC.

Going with the CC version of `DefaultFSErrorHandler` passes the four tests mentioned above ... however, then `DefaultFSErrorHandlerTest` fails b/c it directly tests `DefaultFSErrorHandler` and does not expect that to kill the jvm.

I'm going to go ahead and make one more alternate attempt with this ... I am fairly confident that with one additional change to `JVMStabilityInspector` from OSS C*, we can keep the OSS C* version of `DefaultFSErrorHandler`.

I did find a lingering problem in `FailingRepairTest`, but this seems to be known already in CASSANDRA-18366 on 4.0.

I am also unable to test `CompactionOutOfSpaceTest` locally (it's a ByteMan test, but I've run those before), so I'll see what the build result shows.

Reply:
The specific addition I took from OSS C* `JVMStabilityInspector` that we seemed to be missing in CC (and led to the alternate fix in `DefaultFSErrorHandler`) is this bit in `defaultGlobalErrorHandler`:

```
        if (DatabaseDescriptor.getDiskFailurePolicy() == Config.DiskFailurePolicy.die)
            if (t instanceof FSError || t instanceof CorruptSSTableException)
                isUnstable = true;
```